doc/plugin_agent_nodeattestor_gcp_instance_identity_token.md

```diff
@@ -0,0 +1,11 @@
+# Server plugin: NodeAttestor "gcp_instance_identity_token"
+
+*Must be used in conjunction with the server-side gcp_instance_identity_token plugin*
+
+The `gcp_instance_identity_token` plugin automatically attests instances using the [GCP Instance Identity Token](https://cloud.google.com/compute/docs/instances/verifying-instance-identity). It also allows an operator to use GCP Instance IDs when defining SPIFFE ID attestation policies. 
+
+| Configuration       | Description                                                                                         | Default                                                                       |
+|---------------------|-----------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------|
+| trust_domain        | The trust domain that the agent belongs to.                                                         |                                                                               |
+| audience            | Audience parameter used inside the tokens. Needs be equal to the audience parameter in the server.  |                                                                               |
```

Comment:
We can set a default value but I don't think we should hardcode it. If you run multiple Spire systems on GCP and all their audience settings are the same a IIT can be reused between Spire systems.

Reply:
Hmm that's an interesting point. If we use audience in that fashion, it essentially becomes a secret, because anyone who knows it can just cut a token elsewhere with that audience set. In cases where a valid identity is sufficient to access some services (e.g. env-wide services), this sounds like it could be a serious problem.

Relying on a secret value for audience feels pretty weak, especially because all nodes must have it set and we don't have a good way to protect it like we would a "normal" secret. What if instead, server node attestor plugin has a whitelist of project IDs? Feels like these would change infrequently, and would sufficiently solve for the problem at hand.

Note: the AWS attestor does not suffer from this problem because it calls into the AWS API with configured credentials

Reply:
I think that a whitelisting of projectIDs would be a good starting point. A limitation of this approach would be that you are unable to run multiple SPIRE trust zones inside a single project.